modules/brew/module.yml

```diff
@@ -0,0 +1,23 @@
+name: brew
+shortdesc: The brew module installs Homebrew (Brew) at build time and ensures the package manager remains updated and maintained.
+readme: https://raw.githubusercontent.com/blue-build/modules/main/modules/brew/README.md
+example: |
+  type: brew
+  # List of Brew packages to be installed
+  packages:
```

Comment:
I tested the image.

When using `type: brew`, it works nicely, as it should.

`brew doctor` reports no errors & packages install successfully.

However, when I try to install packages in the image inside `packages:` array, it doesn't work and fails with this error:
`#46 14.06 su: user linuxbrew does not exist or the user entry does not contain all the required fields`

Build log:
https://github.com/fiftydinar/gidro-os/actions/runs/9274948443/job/25518506501?pr=26#step:2:9056

Reply:
@mecattaf Apologize, I removed the comment & migrated it here.

I'll quote you:

> Maybe we should not install brew packages during the build stage. This module can behave like the [flatpak module](https://github.com/blue-build/modules/blob/main/modules/default-flatpaks/user-flatpak-setup?rgh-link-date=2024-05-28T19%3A31%3A03Z): install and update the brew packages listed in recipe.yml while the device is running.

I don't know how it works for Bluefin, I would be interested to investigate.

Reply:
Just checked it, Bluefin doesn't install brew packages in build-time yet.

I think this happens because user is technically not created, as doing mkdir to `/var/home/linuxbrew/` & setting some permissions is not enough.

I'm not sure if we should implement installing packages when system is booted, effectively making it a boot-time module.

I'll think about how to solve this in build-time, but I'm not sure if it's possible.

Reply:
I disabled installing brew packages in build-time until this is figured out.

Reply:
We won't implement installing brew packages for now.

It can be always added in the future if possible & if desired.